### PR TITLE
Upgrade retrofit

### PR DESCRIPTION
**Pull request by Bloodyhacker**

### CODE REVIEW

File: src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java [hunk 1]
```diff
@@ -1,64 +1,90 @@
...
+        httpClient.addInterceptor( loggingInterceptor );
+
+        retrofit = new Retrofit.Builder()
+                .baseUrl( "https://loom.auth0.com/api/v2/" )
+                .client( httpClient.build() )
+                .addConverterFactory( new LoomJacksonConverterFactory() )
+                .addCallAdapterFactory( new LoomCallAdapterFactory() )
                 .build();
-        auth0ManagementApi = adapter.create( Auth0ManagementApi.class );
+        auth0ManagementApi = retrofit.create( Auth0ManagementApi.class );
```
Comment (hochung) [inline src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java]: any reason to not use `RetrofitFactory.newClient`?

Comment (Bloodyhacker) [inline src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java]: Nice catch. I will add this into `RetrofitFactory`

File: src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java [hunk 2]
```diff
@@ -1,64 +1,90 @@
...
+            Auth0UserBasic auth0UserBasic = parsingToPOJO( jsonObjects[ i ] );
             res.put( auth0UserBasic.getUserId(), auth0UserBasic );
         }
         return res;
     }
 
     public Auth0UserBasic getUser( String userId ) {
         JSONObject jsonObject = auth0ManagementApi.getUser( userId );
+        if ( jsonObject == null ) {
+            throw new BadRequestException( "User does not exist!" );
```
Comment (Bloodyhacker) [inline src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java]: Make sense. I think it's time to turn on `@ControllerAdvice`

Comment (hochung) [inline src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java]: Sure, am working on that already.

Comment (geekbeast) [inline src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java]: We also have some error handling logic in the CallAdapter.

File: src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java [hunk 3]
```diff
@@ -1,64 +1,57 @@
...
+        }
         return parsingToPOJO( jsonObject );
     }
 
-    private Auth0UserBasic parsingToPOJO( JSONObject jsonObject ){
+    private Auth0UserBasic parsingToPOJO( JSONObject jsonObject ) {
```
Comment (geekbeast) [inline src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java]: You don't need this anymore just return an Auth0UserBasic in your retrofit interface.

File: src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java [hunk 4]
```diff
@@ -1,64 +1,57 @@
...
-                .setLog( msg -> logger.debug( msg.replaceAll( "%", "[percent]" ) ) )
-                .build();
-        auth0ManagementApi = adapter.create( Auth0ManagementApi.class );
+        retrofit = RetrofitFactory.newClient( "https://loom.auth0.com/api/v2/", () -> token );
+        auth0ManagementApi = retrofit.create( Auth0ManagementApi.class );
     }
 
     public Map<String, Auth0UserBasic> getAllUsers() {
         Map<String, Auth0UserBasic> res = Maps.newHashMap();
         JSONObject[] jsonObjects = auth0ManagementApi.getAllUsers();
```
Comment (geekbeast) [inline src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java]: `Set<Auth0UserBasic> users = auth0ManagementApi.getAllUsers()`

File: src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java [hunk 5]
```diff
@@ -1,64 +1,57 @@
...
 
         for ( int i = 0; i < jsonObjects.length; i++ ) {
```
Comment (geekbeast) [inline src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java]: `return res.stream().collect( Collectors.toMap( Auth0UserBasic::getUserId() , Functions.identity() );`

**Pull request merged**